### PR TITLE
Mylin/#133 spectral line query

### PR DESCRIPTION
**Pull request by acdo2002**
Finish #133

### CODE REVIEW

**Comment by kswang1029** [comment]
Please refer to the comments in the design doc. @acdo2002

**Review by ajm-asiaa — CHANGES_REQUESTED** [review]
I tried SPECTRAL_LINE_QUERY.test.ts and SPECTRAL_LINE_QUERY_INTENSITY_LIMIT.test.ts. 
Both work on MacOS.

The only thing is, it took me a while to find where the `HD163296_13CO_2-1.fits` image is in order to copy over to the MacOS system.

On the 'carta' server it is in `/scratch/images/set_HD163296`
I was expecting it to be somewhere in `/scratch/images/set_QA`
I know you don't have root permissions to move those files around, but maybe we should put `HD163296_13CO_2-1.fits` in `set_QA`. 
In fact, I have just done it:
`/scratch/images/set_QA/HD163296_13CO_2-1.fits`
Would you be able to read it from there instead?

**Comment by zarda** [comment]
> I tried SPECTRAL_LINE_QUERY.test.ts and SPECTRAL_LINE_QUERY_INTENSITY_LIMIT.test.ts.
> Both work on MacOS.
> 
> The only thing is, it took me a while to find where the `HD163296_13CO_2-1.fits` image is in order to copy over to the MacOS system.
> 
> On the 'carta' server it is in `/scratch/images/set_HD163296`
> I was expecting it to be somewhere in `/scratch/images/set_QA`
> I know you don't have root permissions to move those files around, but maybe we should put `HD163296_13CO_2-1.fits` in `set_QA`.
> In fact, I have just done it:
> `/scratch/images/set_QA/HD163296_13CO_2-1.fits`
> Would you be able to read it from there instead?

There are some images for testing moments, such as HD163296_CO_2_1.fits, HD163296_CO_2_1.image, and HD163296_CO_2_1.hdf5. Please @ajm-asiaa kindly move these to /set_QA as well. And I will revise the testing later. Thanks.

**Comment by ajm-asiaa** [comment]
> > I tried SPECTRAL_LINE_QUERY.test.ts and SPECTRAL_LINE_QUERY_INTENSITY_LIMIT.test.ts.
> > Both work on MacOS.
> > The only thing is, it took me a while to find where the `HD163296_13CO_2-1.fits` image is in order to copy over to the MacOS system.
> > On the 'carta' server it is in `/scratch/images/set_HD163296`
> > I was expecting it to be somewhere in `/scratch/images/set_QA`
> > I know you don't have root permissions to move those files around, but maybe we should put `HD163296_13CO_2-1.fits` in `set_QA`.
> > In fact, I have just done it:
> > `/scratch/images/set_QA/HD163296_13CO_2-1.fits`
> > Would you be able to read it from there instead?
> 
> There are some images for testing moments, such as HD163296_CO_2_1.fits, HD163296_CO_2_1.image, and HD163296_CO_2_1.hdf5. Please @ajm-asiaa kindly move these to /set_QA as well. And I will revise the testing later. Thanks.

OK @zarda I have moved them:
/scratch/images/set_QA/HD163296_13CO_2-1.fits
/scratch/images/set_QA/HD163296_CO_2_1.fits
/scratch/images/set_QA/HD163296_CO_2_1.image
/scratch/images/set_QA/HD163296_CO_2_1.hdf5

**Comment by acdo2002** [comment]
@kswang1029 I add `SPECTRAL_LINE_QUERY_INTENSITY_LIMIT.test.ts` & `SPECTRAL_LINE_QUERY_WIDE_FREQ.test.ts ` in order to fit your suggestions. The first one tests the different intensity limits, the second one test request a wide freq range to check the upper data transferred limit: 100000

@ajm-asiaa Hi Anthony, 
Thanks to copy the images, I have changed the testdirectory to set_QA in the SPECTRAL_LINE_QUERY*.test.ts.

@zarda @ajm-asiaa 
Since the Issue #149 has been solved (many thanks to Heng-tai, he found the new websocket lib), the SPECTRAL_LINE_QUERY related codes have been finished. The ws & wss are both working fine, please review them, thanks!!!

**Review by ajm-asiaa — APPROVED** [review]
Looks good! All 3 tests seem to pass fine on MacOS.
SPECTRAL_LINE_QUERY_WIDE_FREQ.test.ts
SPECTRAL_LINE_QUERY_INTENSITY_LIMIT.test.ts
SPECTRAL_LINE_QUERY.test.ts

File: src/test/config.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "serverURL": "wss://carta.asiaa.sinica.edu.tw/socketdev",
+    "serverURL": "wss://carta.asiaa.sinica.edu.tw/socketicd",
```
Comment (zarda) [inline src/test/config.json:2]: Better to be /socketdev

**Review by zarda — APPROVED** [review]

File: src/test/config.json (hunk 1 above)
Comment (acdo2002) [reply src/test/config.json:2]: Thanks!!! I forget to change back to the socketdev!!!

**Pull request merged**